elastic/apm-server
# Set default transaction and span type if empty

## Motivation/summary

When an agent sends an event with `transaction.type` or `span.type` set to an empty string, have the server set a default value of "unknown".

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

Using the Go agent, send a transaction and span without type specified:

```go
tx := tracer.StartTransaction("name", "")
span := tx.StartSpan("name", "", nil)
span.End()
tx.End()
```

The transaction and span should be indexed with `transaction.type: unknown` and `span.type: unknown` respectively.

## Related issues

Closes #7376
